### PR TITLE
Add hazelcast-full-example.yaml

### PR DESCRIPTION
**Pull request by blazember**
Adding `hazelcast-full-example.yaml` to the `3.12-BETA-1` release.
Cherry-picked from the master PR: https://github.com/hazelcast/hazelcast/pull/14557

### CODE REVIEW

File: hazelcast/src/main/resources/hazelcast-full-example.yaml [hunk 1]
```diff
@@ -0,0 +1,2515 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This is a full example hazelcast.yaml that includes all the
+# configuration elements and attributes of a Hazelcast member.
+#
+# To use this, rename it to hazelcast.xml and place it in
```
Comment (tkountis) [inline hazelcast/src/main/resources/hazelcast-full-example.yaml]: xml -> *yaml*

File: hazelcast/src/main/resources/hazelcast-full-example.yaml [hunk 2]
```diff
@@ -0,0 +1,2515 @@
...
+#
+# To learn how to configure Hazelcast, please see the Reference Manual
+# at https://hazelcast.org/documentation/
+
+hazelcast:
+
+  # You can use import to load different Hazelcast declarative configuration files you prepared.
+  # You can import as many YAML files as you want and hence compose your Hazelcast configuration
+  # out of those YAML files. If you want to use import, it should be placed at the top of your
+  # Hazelcast YAML file right after the hazelcast node.
```
Comment (tkountis) [inline hazelcast/src/main/resources/hazelcast-full-example.yaml]: Hazelcast(cap 1st letter)

Comment (blazember) [reply hazelcast/src/main/resources/hazelcast-full-example.yaml]: Here it's the name of the YAML configuration root node. 
So what about this:
`... after the "hazelcast" root node` ?

File: hazelcast/src/main/resources/hazelcast-full-example.yaml [hunk 3]
```diff
@@ -0,0 +1,2515 @@
...
+  # symmetric-encryption:
+  #   enabled: true
+  #   algorithm: PBEWithMD5AndDES
+  #   salt: thesalt
+  #   password: thepass
+  #   iteration-count: 19
+  # * "member-address-provider":
+  # IMPORTANT
+  # This configuration is not intended to provide addresses of other cluster members with
+  # which the hazelcast instance will form a cluster. This is an SPI for advanced use in
```
Comment (tkountis) [inline hazelcast/src/main/resources/hazelcast-full-example.yaml]: cap H for Hazelcast

File: hazelcast/src/main/resources/hazelcast-full-example.yaml [hunk 4]
```diff
@@ -0,0 +1,2515 @@
...
+  #     - "class-name":
+  #         The name of the class implementing the com.hazelcast.spi.MemberAddressProvider interface.
+  #     - "properties":
+  #         The properties that will be provided when constructing the provided MemberAddressProvider. Hazelcast will
+  #         first try instantiating the provided class by invoking a constructor accepting a single
+  #         java.util.Properties instance. In the case where there is no such constructor and there are also
+  #         no properties defined by this configuration, Hazelcast will exceptionally try to use the no-arg
+  #         constructor.
+  #
+  network:
```
Comment (tkountis) [inline hazelcast/src/main/resources/hazelcast-full-example.yaml]: missing rest-api?

Comment (tkountis) [reply hazelcast/src/main/resources/hazelcast-full-example.yaml]: found it

Comment (blazember) [reply hazelcast/src/main/resources/hazelcast-full-example.yaml]: It's in line 623.

File: hazelcast/src/main/resources/hazelcast-full-example.yaml [hunk 5]
```diff
@@ -0,0 +1,2515 @@
...
+      capacity: 100
+      quorum-ref: quorumRuleWithThreeNodes
+      merge-policy:
+        batch-size: 100
+        class-name: PutIfAbsentMergePolicy
+
+  #
+  # ===== HAZELCAST CARDINALITY ESTIMATOR SERVICE CONFIGURATION =====
+  #
+  # Configuration element's name is "cardinality-estimator-service". It has the optional attribute "name" with which you
```
Comment (tkountis) [inline hazelcast/src/main/resources/hazelcast-full-example.yaml]: element name different that the actual element

File: hazelcast/src/main/resources/hazelcast-full-example.yaml (hunk 2 above)
Comment (tkountis) [reply hazelcast/src/main/resources/hazelcast-full-example.yaml]: you are right, i mis-read it.

**Pull request merged**